### DISCUSSION TITLE
Page Template > combine x Elements in one section

### DISCUSSION THREAD

**Discussion by normenrollf** [sulu/sulu#6687] [Q&A]
Is it possible to combine individual blocks into one block? Currently, each block element gets its own section. Since we store many options in the templates, it is sometimes annoying that each element acts separately from the next element. 

Example: Element 1: Additional headline, Element 2: Text, Element 3: Image gallery, Element 4: Articles preview of the last 3 articles.

The section, for example, gets a padding of 100px 0. The element AdditionalHeadline and the text then also have a padding of 100px 0. It just looks stupid. 

Thanks in advance

**Comment by alexander-schranz** [comment]
Hello @normenrollf,

I'm not sure what you mean. Sulu is not shipping any blocks or forcing you anything here which fields you need to use. The title and url fields are the only required fields in your templates for sulu.

So you can combine different fields in blocks and combine them the way you want it. Create one block with several fields or create a multiple different blocks. All is in your control and how it looks on the website is also in 100% in your control. So create the blocks in your xml templates they way which best matches your usecase and which best supports the workflow of you and your content managers.

I must say I would not recommend to create "single elements" blocks as that make it harder for content manager to create content as he always need to do several clicks. We mostly have a default `text-with-image` block with `title`, `description`, `image` field, see the demo as example. But at the end you are free to structure it the way which matches your case the best.

If you want to group all blocks together, as example all blocks which has not its own headline, you can just create a twig extension in your project: https://symfony.com/doc/current/templating/twig_extension.html. This way you can use PHP to group the things together.

But as said sulu isn't forcing here anything, its all in your control the html, css and template structure. So you can even use something like `.block-text + .block-image` has different spacing between. Or checking in your twig file `previousBlockType` and `currentBlockType` to add another class. So several ways how you can control spacing between your blocks.

**Reply by normenrollf** [reply]
Hello Alexander, may I send you a direct Email incl. screenshot and in german?

**Comment by ajdinmore** [comment]
I think what @normenrollf may be after here is almost achieved by having all page block types as child blocks of a root "group" block type.

IIRC, it won't be possible to move blocks between groups then, which may be a problem (less so in 2.5?)

**Reply by alexander-schranz** [reply]
I personally would try to solve it as said on the twig part here. So you control it and don't move "presentation" specific things here to the content manager. It better to have things in control of the code instead of content manager to keep things consistent through the whole website. But I understand if somebody want to use here nested blocks.

**Comment by mbaderweb** [comment]
nested Blocks would be an incredible feature. So you can minimalize the blocks and reuse it in any way

**Reply by alexander-schranz** [reply]
Nested blocks are already supported by Sulu since 2.1.0: https://sulu.io/blog/sulu-release-2-1-0-2-0-9

**Reply by mbaderweb** [reply]
Great!!! Thank you!
